src/ReverseProxy/Configuration/ConfigurationConfigProvider.cs

```diff
@@ -182,9 +182,37 @@ private void Convert(ProxyMatch proxyMatch, ProxyMatchData data)
             proxyMatch.Methods = data.Methods?.ToArray();
             proxyMatch.Hosts = data.Hosts?.ToArray();
             proxyMatch.Path = data.Path;
+            proxyMatch.Headers = Convert(data.Headers);
         }
 
-        private CircuitBreakerOptions Convert(CircuitBreakerData data)
+        private static IReadOnlyList<RouteHeader> Convert(IReadOnlyList<RouteHeaderData> headers)
+        {
+            if (headers == null)
+            {
+                return null;
+            }
+
+            return headers.Select(data => Convert(data)).ToArray();
```

Comment:
Can we reduce it to the single line?
```suggestion
            return headers?.Select(data => Convert(data)).ToArray();
```